hydro_lang/src/ir.rs

```diff
@@ -2501,4 +2674,52 @@ mod test {
     fn hydro_leaf_size() {
         insta::assert_snapshot!(size_of::<HydroLeaf>(), @"160");
     }
+
+    #[test]
+    fn test_simplify_q_macro_basic() {
+        // Test basic non-q! expression
+        let simple_expr = "x + y";
```

Comment:
We can go a step further and actually use `q!(...)` here for the tests and splice it in various ways (`.splice_untyped_ctx`, `.splice_typed_ctx`, `.splice_fn*_ctx`) so that we can make sure the extraction stays in-sync with Stageleft.

Reply:
cleaned up.